### ISSUE TITLE
Create ride

### ISSUE THREAD

**Issue by carduz** [sweng2/sweng2#56]
We said that the driver create the ride but if we have a lot of passenger we have more than one driver. How do we proceed in this case?

**Comment by gdejaege** [comment]
I do not remember exactly what we said about it but I think we said that if there are to many passengers, the request is split in two requests, so two drivers should have to build the ride.

**Comment by tux-mind** [comment]
Ooops.

that's a logic leak :confused: 

in our model the `Ride` have many `Driver`s, but there is no way to build a `Ride` with more `Driver`s.

the "split" solution can work but we have to copy the client to the new requests.

ATM i have no good alternative for the problem, simply because we could not satisfy the minimum requirements for building the `Ride` ( N `Driver`s accepted the request in K minutes ), so waiting for the completion of the requirements should be highly unproductive.

splitting the request will not solve the problem, but will make it easier, except for all other stuff that comes in like notification to the client ( he may already leave with the first `Ride`, and we cannot contact other guys to tell them anything ).

As a user I think that if I order a ride I want to start and arrive with all my friends, not only a few.

suggestions ?

**Comment by carduz** [comment]
the problem is for sharing ride

**Comment by gdejaege** [comment]
I think the split solution is not a magic solution but should be enough in the majority of cases. When A client is reserving a taxi for then persons for example, the two or three requests are proposed to the taxi queue. The taxi drivers are supposed to give their answer relatively fast. If the users have to wait a long time even before the taxi accepts, and then also wait long for the taxi that have to arrive, the delay will be to long. 

Therefore in the majority of the cases the client should receive his notifications before the first taxi arrives.

**Comment by carduz** [comment]
but in this way we have to change a lot of things, I think that we have not time to do that

**Comment by carduz** [comment]
we have to change RASD and DD

**Comment by tux-mind** [comment]
I agree with you @gdejaege , it's ok for me to use the split option.
as @carduz said we have to update the RASD and DD, but have no fear, it's better to add a hotfix rather than leaving a known bug :wink:

**Comment by carduz** [comment]
We can say that if a client perform a request for more than 4 people, all taxi needed have to accept the request. They go to the client together.
In real applications this is the approach: the company find taxies available and then send them to the client

**Comment by carduz** [comment]
however I think that the best solution is split the request and copy the client. what do you think? I think that in this way we have only to add some requirements in RASD, without change other things

**Comment by carduz** [comment]
done (as requirement in RASD), but keep this for the next tasks
